src/backend/common/data/commands/auth/permissions.py

```diff
@@ -14,59 +14,42 @@ class CheckUserHasPermissionCommand(Command[bool]):
     tournament_id: int | None = None
 
     async def handle(self, db_wrapper, s3_wrapper):
-        series_id = self.series_id
-
         async with db_wrapper.connect() as db:
-            denied_permission_exists = False
             def check_perms(rows: Iterable[Row]):
                 num_rows = sum(1 for r in rows) # type: ignore
                 if num_rows == 0:
-                    # if check_denied_only is True, we only care about the absence of a denied permission,
-                    # so an empty result set satisfies that.
-                    # if we have previously found a denied permission, we must have a explicitly
-                    # approved permission to override it, so don't return true in that case
-                    if self.check_denied_only and not denied_permission_exists:
-                        return True
-                    else:
-                        return False
-                # if we find an instance of the permission which isnt denied, we have the permission no matter what,
-                # so just return True once we find one. otherwise, the permission must have been denied, so we return
-                # False after iterating
+                    return None
+                # if we find an instance of the permission which is denied, always just return False
                 for row in rows:
                     is_denied = row[0]
-                    if not is_denied:
-                        return True
-                return False
+                    if is_denied:
+                        return False
+                return True
             
-            if self.tournament_id:
-                # if tournament is part of a series, we want to find out its id so we can check series permissions also
-                if not series_id:
-                    async with db.execute("SELECT series_id FROM tournaments WHERE id = ?", (self.tournament_id,)) as cursor:
-                        row = await cursor.fetchone()
-                        if not row:
-                            raise Problem("Tournament not found", status=400)
-                        series_id = row[0]
+            #finally, check user roles
+            async with db.execute("""
+                SELECT DISTINCT rp.is_denied
+                FROM roles r
+                JOIN user_roles ur ON ur.role_id = r.id
+                JOIN role_permissions rp ON rp.role_id = r.id
+                JOIN permissions p on rp.permission_id = p.id
+                WHERE ur.user_id = ? AND p.name = ?
+                """, (self.user_id, self.permission_name)) as cursor:
+                rows = await cursor.fetchall()
+
+            perm_check = check_perms(rows)
+            if perm_check is not None:
```

Comment:
I think with this, if there is a deny in the series/tournament/team roles, it won't be factored in even though it should?